service/raft/raft_group0.cc

```diff
@@ -298,6 +298,7 @@ future<> raft_group0::join_group0() {
         auto timeout = db::timeout_clock::now() + std::chrono::milliseconds{1000};
         netw::msg_addr peer(raft_addr_to_inet_addr(g0_info.addr));
         try {
+            // TODO: aborts?
```

Comment:
Historically raft aborts were added after this code was written so these todo items are valid.

Reply:
You can check `git` to see who added it and I do not get timed part.

Reply:
TODO is a note to self. As a note to others it lives in github. Giving someone else a todo item is not a good idea. So this is why todo items should be personalized and timed. It's a commitment of the author of the patch to do something. Otherwise it has little value, just wishful thinking which sits and rots in the code since nobody ever deletes it.

Reply:
Well, for me TODO is a reminder that something should be done sometime by someone. If something needs to be done it is not a little value to note it in the code. Sometimes simpler than open an issue. Never ever saw a timed TODO in any code.

Reply:
Here is a good discussion of TODO  and FIXMEs:

https://wiki.c2.com/?FixmeComment

I am more from the extreme programming camp, so this reflects best what I think about it:
__
I would recommend giving up the FIXME and TODO comments completely. If there is a problem, fix it. If you don't have time to fix it now, you give up the right to specify to someone else how to fix something. If something is required functionality, implement it now. If something is functionality you just thought up, ignore it until it is asked for. It is quite arrogant to place demands on the follow up programmers, so don't do it. -- [WayneMack](https://wiki.c2.com/?WayneMack)
---

Reply:
> For a small project what you write would be true perhaps. For a larger project this duplicates the issue tracker, with no clear responsibility or dates. Moreover, some people tend to leave more todos than others. As a result, you get a random, non-systematic littering of the code with obsolete comments with no clarity or process for removing them. Older code bases contains thousands of todo items.

Sometimes it is much easier to point to a small issue you left in the code as a TODO then open an issue about it immediately after your code is committed (which in our case may take a while so you will probably forget). Also if you think nobody looks at TODOs, do you think anybody will look at those issues? It is much higher chance that whoever needs to touch your code will fix the TODO then for someone randomly choosing to fix your issue. May be in a real OS with a lot of contributors, but this is not our case.  

I do not see how you arrive to "obsolete" conclusion and why there is not clarity how to remove them. You remove them when you fix them.

That said XXX comments should be removed :)

Reply:
> Reduction to absurdity is a logical fallacy . https://en.wikipedia.org/wiki/Reductio_ad_absurdum

It's not a fallacy, it's a valid method for arguing, it's even used in mathematics quite often to prove things.

>  I am amazed you mentioned github since I was quite explicit that github is a more proper place for this than the code base.

Yes issues should be created, but from my experience there is maybe one person in the whole company who opens issues after spotting them while doing something in the code (and it's @nyh) :)

IMO:
github issue + fixme > just github issue or just fixme > nothing
where ">" means "is better than"
I'm not sure if "just github issue" is comparable with "just fixme", issue is probably better, but I'd prefer to have a fixme which points me into the issue anyway.

>  My take is that this patch is overusing TODO and FIXME items, just please consider it and try to address.

I agree, I'll remove some of them.